src/Control/Effect/Sum.hs

```diff
@@ -16,14 +16,15 @@ import Data.Kind (Constraint)
 import GHC.Generics (Generic1)
 
 -- | Higher-order sums are used to combine multiple effects into a signature, typically by chaining on the right.
-data (f :+: g) (m :: * -> *) k
-  = L (f m k)
-  | R (g m k)
+data (l :+: r) (m :: * -> *) k
+  = L (l m k)
+  | R (r m k)
   deriving (Eq, Foldable, Functor, Generic1, Ord, Show, Traversable)
 
 infixr 4 :+:
 
-instance (Effect f, Effect g) => Effect (f :+: g)
+instance (Effect l, Effect r) => Effect (l :+: r) where
+  type CanHandle (l :+: r) f = (CanHandle l f, CanHandle r f)
```

Comment:
An earlier iteration of this had given `CanHandle` as

```haskell
type CanHandle sig :: (* -> *) -> Constraint
```

(Note the partial application.) That signature required us to define a new class for `Sum`’s type instance:

```haskell
class (cl f, cr f) => (cl & cr) (f :: * -> *)
```

This worked astonishingly well, but it’s honestly more flexible to just fully apply it, so it is no more. But I wanted to mention this because that sort of possibility really makes me love Haskell.